common/brave_channel_info.h

```diff
@@ -9,6 +9,8 @@
 #include <string>
 
 namespace brave {
+// DEPRECATED: Use `version_info::GetChannelString(chrome::GetChannel())`
```

Comment:
but this can't be used without copypasting `GetBraveChannel`, right? we shouldn't urge people to refactor code without providing a good alternative. We just need a better utility

Reply:
cc @bridiver

Reply:
@iefremov why do we need a utility? This is how you're supposed to get the channel string and there are many usages of it in upstream. If they don't need a utility, why do we?

Reply:
@bridiver I'm saying that copypasting 15 lines of code is not good and we should wrap it into something. Not sure if chrome does something similar

Reply:
@iefremov in terms of the mapping of `stable` -> `release`, we definitely do not want a helper for that because we don't want people to use `release`. These are exceptions for backwards compat because it's too complicated to change it server-side at this point. @DJAndries will add comments and @tmancey is also going to make the change to ads in this PR so we can completely remove this method

Reply:
we also discussed creating a component with `visibility` restricted and name it something like `deprecated_channel_name`. We definitely don't want to leave anything here because it's also causing a layering violation in ads